test/validate/windows_validate.go

```diff
@@ -130,3 +156,40 @@ func azureVnetIpamIps(result []byte) (map[string]string, error) {
 	}
 	return azureVnetIpamPodIps, nil
 }
+
+func validateHNSNetworkState(ctx context.Context, nodes *corev1.NodeList, clientset *kubernetes.Clientset, restConfig *rest.Config) error {
+	// check windows HNS network state
+	for index := range nodes.Items {
+		pod, err := acnk8s.GetPodsByNode(ctx, clientset, privilegedNamespace, privilegedLabelSelector, nodes.Items[index].Name)
+		if err != nil {
+			return errors.Wrap(err, "failed to get privileged pod")
+		}
+
+		podName := pod.Items[0].Name
+		// exec into the pod to get the state file
+		result, err := acnk8s.ExecCmdOnPod(ctx, clientset, privilegedNamespace, podName, hnsNetworkCmd, restConfig)
+		if err != nil {
+			return errors.Wrap(err, "failed to exec into privileged pod")
+		}
+
+		hnsNetwork, err := hnsNetworkState(result)
+		if err != nil {
+			return errors.Wrap(err, "failed to unmarshal HNS state file")
+		}
+
+		// check hns properties
+		if len(hnsNetwork) == 1 {
+			return errors.New("HNS default ext network or azure network does not exist")
```

Comment:
Would be nice to log the `hnsNetwork` as the error says it is missing one of those info.